### PR TITLE
15 display all attachment types

### PR DESCRIPTION
**Pull request by nmmull**
Fairly straightforward. Just a few notes:
* I added *very* simple flag handling in the executable. As we add more flags, we will definitely want to use one of the packages that Matt has mentioned.
* I tested on a couple of the `mbox`s we have access to, seems to work alright, a bit slow, but I don't think it could be terribly faster unless it was parallelized.

### CODE REVIEW

**Review by bufordrat — CHANGES_REQUESTED** [review]
This is awesome.  One small thing: I got this output on one of my tests:

```console
$ dune exec -- ./main.exe --report < path/to/matts/example/Sent.mbox
Attachment Types:    
        text/plain
  application/msword
  application/octet-stream
  application/pdf
  application/rtf
  application/vnd.ms-excel
  application/vnd.ms-powerpoint
  image/gif
  image/jpeg
  text/html
  text/plain

```
Not sure off the top of my head what causes the extra indented `text/plain` at the beginning, but if an easy fix suggests itself, might be worth doing.  Does whitespace need to get trimmed off the MIME type strings?  Not sure.  I can provide the MBOX that does this if you can't replicate it.

One other example:

```console
$ dune exec -- ./main.exe --report < path/to/matts/example/In-7-02.mbox
Attachment Types:    
  Text/Plain
  application/octet-stream
  application/rtf
  text/plain
  text/x-vcard
```
The double `text/plain` situation suggests it might be a good idea to downcase all the MIME strings before deduping them.

**Comment by waclena** [comment]
Matt Teichman ***@***.***> writes:

> Not sure off the top of my head what causes the extra indented
> text/plain at the beginning, but if an easy fix suggests itself, might
> be worth doing. Does whitespace need to get trimmed off the MIME type
> strings? Not sure.

Mime headers in parts are exactly like normal mail headers at the top of
a message.  In the latter case, there can be any amount of whitespace
separating the header name from the value, so the same is true for
headers in mime parts.  So the value part of these two headers:

    Content-Type: text/plain
    Content-Type:                  text/plain

should be the same: "text/plain".

I'd expect the ocamlnet header parser to return it correctly, but if it
doesn't I think we should String.(trim whitespace) the value.

File: main.ml [hunk 1]
```diff
@@ -25,7 +31,18 @@ let () =
        match converted_email with
        | Error err    -> print (Lib.Error.message err)
        | Ok converted -> write stdout converted
-  else print (Printf.sprintf "Error: missing config file '%s'" default_config_name)
+  else Printf.printf "Error: missing config file '%s'\n" default_config_name
+
+(* A _very_ minimial executable *)
```
[the remark this answers is not in the document]
Comment (nmmull) [reply main.ml]: Noted, fixed. Thanks for pointing it out.

**Comment by nmmull** [comment]
Yeah, admittedly what I'm doing is super basic, I wanted to see if we could get away without using the ocamlnet parser.

But I think the issue was just a dumb one, I was trimming spaces instead of all whitespace. Fixed this, I probably won't be able to test it until Wednesday but since it was such a simple fix, I pushed it and you can review/test it before then if you have time.

**Pull request merged**